[DOCS] Add contributing to README, and disable vscode autoformat

bradzacher opened the pull request:
The vscode autoformatter is completely at odds with prettier, so if you don't have the prettier extension installed, you end up with wild changes.

Additionally if you've got both the beautify and prettier extensions installed, it can cause some serious issues.

Forcing it to be on will probably cause a lot of headaches to anyone trying to contribute unless they get their environment setup perfectly (sorry @weirdpattern for earlier!!)

weirdpattern commented on .editorconfig:
```diff
@@ -6,3 +6,7 @@ indent_size = 4
 insert_final_newline = true
+
+# yarn / npm uses 2 spaces when it dumps the file after a change
+[package.json]
```
Given that .eslintrc is a json file, should we include it here too for consistency?

bradzacher replied:
The rest of the codebase is 4 spaces!
The problem is that when you do yarn / npm install with old versions, they'll reformat the file to 2 spaces.

This is just to try to stop future vscode autoformat on save problems.

weirdpattern replied:
Ah, that makes sense... let's forget about this

weirdpattern commented on README.md:
```diff
@@ -76,3 +76,24 @@ Then configure the rules you want to use under the rules section.
+
+## Contributing
+
+For a PR to be merged, all tests must pass, the must be no lint errors, and the code must be formatted.
```
`there must be`... instead of `the must be`

weirdpattern commented on README.md:
```diff
@@ -76,3 +76,24 @@ Then configure the rules you want to use under the rules section.
+- `yarn test`
+- `yarn lint`
+- `yarn format`
+There is a commit hook which will will help you follow this.
```
`will` instead of `will will`

weirdpattern commented on README.md:
```diff
@@ -76,3 +76,24 @@ Then configure the rules you want to use under the rules section.
+## Contributing
+
+For a PR to be merged, all tests must pass, the must be no lint errors, and the code must be formatted.
+- `yarn test`
```
Technically `yarn test` will run `yarn lint` for you as it has a `pretest` script.
But I'm ok leaving it as is... emphasizes the importance of running both :smile:...

bradzacher replied:
I just changed the config so that doesn't to make it a bit easier to run tests.
Gets a bit annoying when you're developing a rule to have to wait for the lint + docs check as well.

I've changed the both the precommit hook and travis so they explicitly run all of the checks.

(test run with the new config: https://travis-ci.org/bradzacher/eslint-plugin-typescript/jobs/460887254)

weirdpattern commented:
One more thing, now that I've merged #186, you might want to add a new rule to the .gitignore file... I'm excluding everything in the .vscode except for specific files (currently just the settings.json). 

I'm OK with inverting the rule to exclude specific files instead... whatever makes it easier to understand...

weirdpattern commented:
I just thought of something else... what about moving the contributing section from the README.md to its own CONTRIBUTING.md file? it seems to be a common practice and is even suggested in the [insights/community](https://github.com/bradzacher/eslint-plugin-typescript/community) tab

bradzacher commented:
Man github handles rebases weirdly..

I just:
- Moved to separate CONTRIBUTING.md
- Adjusted `yarn test`/precommit/travis so they match the docs better.
  - `yarn test` only runs the tests
  - precommit / travis both now run the lint, format, test and docs checks
